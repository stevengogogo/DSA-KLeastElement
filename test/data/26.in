10 50000
5804 6359 6227 -8213 4855 -8247 8387 8622 -2278 -6316
Reverse 1 8
Reverse 4 5
Query 5 5 1
Delete 2
Insert 10 1424
Insert 4 8546
Delete 4
Reverse 8 8
Delete 2
Delete 9
Insert 5 8491
Delete 1
Query 3 5 3
Insert 4 -6445
Query 2 5 3
Delete 6
Insert 7 -1762
Insert 1 -2396
Query 7 10 1
Query 7 8 1
Insert 6 1110
Query 11 11 1
Insert 11 7106
Delete 7
Delete 6
Reverse 7 10
Reverse 4 6
Delete 4
Delete 7
Query 1 6 4
Query 1 4 1
Insert 3 1343
Insert 4 7256
Insert 1 7833
Query 11 11 1
Delete 4
Insert 2 -1489
Insert 10 276
Query 3 8 6
Query 2 2 1
Delete 1
Insert 12 3049
Insert 10 6216
Reverse 13 13
Insert 4 -7058
Reverse 6 8
Insert 11 -512
Delete 12
Delete 8
Insert 4 1146
Reverse 1 2
Delete 13
Delete 9
Query 3 3 1
Delete 9
Insert 11 8260
Reverse 5 11
Query 4 5 2
Insert 2 -9973
Reverse 13 13
Delete 12
Insert 8 -3434
Insert 9 -7703
Reverse 8 10
Reverse 4 12
Query 3 4 1
Insert 6 -7805
Query 14 14 1
Delete 5
Query 3 5 3
Reverse 5 12
Reverse 10 14
Reverse 4 10
Delete 9
Insert 14 1998
Query 11 12 2
Reverse 1 8
Reverse 11 13
Reverse 9 10
Insert 2 2584
Delete 4
Reverse 13 13
Insert 12 7944
Delete 15
Delete 9
Insert 8 -5041
Query 2 8 6
Insert 13 4724
Delete 1
Insert 13 5797
Insert 11 -3330
Query 6 7 2
Delete 14
Insert 3 -9372
Insert 2 7314
Reverse 9 14
Reverse 5 5
Query 3 4 2
Delete 15
Insert 13 -9321
Insert 18 -9539
Reverse 11 16
Delete 2
Reverse 11 17
Delete 12
Reverse 5 14
Delete 13
Insert 15 -9058
Delete 5
Query 6 12 7
Query 8 14 7
Reverse 13 13
Insert 9 7970
Delete 4
Delete 6
Delete 10
Delete 12
Reverse 12 12
Reverse 7 11
Delete 1
Query 11 11 1
Insert 12 -1572
Reverse 5 11
Reverse 9 10
Reverse 9 12
Delete 12
Query 11 11 1
Insert 8 7186
Query 3 6 4
Reverse 9 9
Query 6 10 1
Reverse 3 11
Reverse 11 11
Delete 6
Delete 4
Insert 3 4751
Query 1 6 4
Insert 1 -7986
Insert 6 -7418
Reverse 6 11
Reverse 3 8
Insert 14 -8631
Delete 3
Reverse 7 9
Query 9 10 1
Delete 5
Insert 12 -7344
Reverse 4 11
Delete 1
Delete 6
Delete 5
Reverse 6 6
Insert 2 9820
Query 11 11 1
Reverse 1 1
Delete 4
Insert 6 7096
Insert 5 -9593
Delete 11
Query 4 7 2
Query 11 11 1
Query 11 11 1
Query 9 9 1
Insert 8 674
Delete 4
Insert 7 3208
Query 5 9 2
Query 1 7 3
Delete 10
Query 7 8 1
Delete 9
Delete 7
Reverse 9 9
Insert 2 652
Query 1 10 8
Reverse 4 4
Delete 6
Insert 9 1363
Reverse 10 10
Reverse 6 6
Delete 3
Delete 5
Query 1 2 2
Insert 6 -6261
Query 8 9 1
Insert 3 9771
Insert 7 -9975
Query 3 6 4
Delete 3
Reverse 10 10
Delete 4
Insert 4 730
Delete 2
Reverse 8 8
Query 8 8 1
Reverse 1 3
Insert 3 -8811
Reverse 5 9
Delete 2
Insert 8 -8457
Delete 2
Query 6 8 3
Query 6 6 1
Reverse 4 9
Query 8 9 2
Delete 2
Query 1 2 2
Insert 4 -5665
Delete 3
Insert 2 21
Delete 7
Reverse 5 8
Query 7 7 1
Delete 5
Delete 5
Insert 4 4827
Insert 4 -2885
Reverse 2 5
Delete 3
Insert 6 -9028
Query 1 5 3
Delete 3
Reverse 7 7
Query 3 5 3
Reverse 5 5
Reverse 3 6
Reverse 4 7
Delete 7
Reverse 6 6
Query 1 2 1
Insert 7 3317
Query 1 2 2
Delete 1
Reverse 4 5
Delete 6
Insert 6 -4665
Delete 3
Query 4 5 2
Insert 1 -6318
Query 5 5 1
Query 6 6 1
Insert 5 -8357
Insert 4 -8673
Reverse 5 8
Reverse 2 7
Reverse 7 8
Delete 7
Delete 7
Reverse 3 6
Reverse 3 4
Reverse 1 6
Reverse 6 6
Query 4 5 2
Query 1 4 2
Delete 3
Insert 5 5721
Reverse 4 6
Query 1 4 3
Delete 4
Insert 2 -2521
Delete 5
Reverse 4 4
Reverse 1 2
Reverse 5 5
Delete 1
Insert 4 4592
Insert 1 -9550
Reverse 5 6
Delete 4
Insert 2 -8510
Insert 6 -5681
Delete 7
Reverse 5 5
Query 4 6 2
Insert 6 -6951
Delete 7
Delete 5
Delete 4
Reverse 3 3
Query 1 2 2
Insert 2 8819
Reverse 2 4
Query 1 5 2
Insert 3 4867
Insert 2 -490
Delete 7
Reverse 1 4
Query 3 4 2
Insert 2 -875
Insert 3 -1807
Reverse 2 8
Query 8 8 1
Reverse 7 8
Delete 7
Query 4 4 1
Insert 8 9914
Reverse 8 8
Query 7 8 2
Reverse 1 8
Delete 3
Insert 7 1054
Delete 7
Reverse 6 6
Reverse 7 7
Delete 3
Query 1 2 2
Delete 6
Query 1 3 3
Reverse 4 5
Delete 2
Delete 2
Insert 4 -8535
Delete 2
Reverse 1 1
Query 3 3 1
Query 3 3 1
Reverse 1 1
Query 3 3 1
Insert 1 -5648
Query 4 4 1
Insert 2 -2126
Delete 1
Reverse 3 3
Reverse 4 4
Delete 3
Query 1 1 1
Delete 2
Insert 2 4909
Reverse 2 3
Delete 1
Insert 1 -730
Delete 2
Delete 2
Delete 1
Insert 1 -98
Reverse 1 1
Delete 1
