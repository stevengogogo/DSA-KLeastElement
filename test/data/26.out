4855
8491
6227
-6316
-1762
-6316
-2396
-8213
-1762
7256
-1489
-8213
8260
-1489
7256
6227
-3434
3049
-5041
8260
7944
7944
-5041
3049
7970
-3330
4751
-9372
-8631
-3330
-8631
-8631
4751
674
-3330
674
7096
8260
-8631
9771
1363
674
-9975
-6261
1363
-8457
730
21
730
4827
21
21
-4665
-8357
-8457
-4665
-5681
-8510
-8510
-490
-875
-9550
9914
9914
9914
-8535
-8535
-8535
-8535
-2126
